core/src/main/java/com/sequenceiq/cloudbreak/job/existingstackfix/ExistingStackPatcherJob.java

```diff
@@ -0,0 +1,112 @@
+package com.sequenceiq.cloudbreak.job.existingstackfix;
+
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.Status;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.StackPatchType;
+import com.sequenceiq.cloudbreak.domain.view.StackView;
+import com.sequenceiq.cloudbreak.quartz.statuschecker.job.StatusCheckerJob;
+import com.sequenceiq.cloudbreak.service.stackpatch.ExistingStackPatchApplyException;
+import com.sequenceiq.cloudbreak.service.stackpatch.ExistingStackPatchService;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+import com.sequenceiq.cloudbreak.service.stack.StackViewService;
+
+import io.opentracing.Tracer;
+
+@DisallowConcurrentExecution
+@Component
+public class ExistingStackPatcherJob extends StatusCheckerJob {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExistingStackPatcherJob.class);
+
+    private static final EnumSet<Status> FAILED_OR_IN_DELETE_STATUSES = EnumSet.of(
```

Comment:
STOP state should also be filtered right?

Reply:
No, these are the statuses where we will not even retry the patching. 
Stop case will be handled by whether the CM server is in reachable state, so if the stack is stopped, we will try to patch it in the next job run.